refactor: change error handling to be strongly-typed

ClaytonKnittel opened the pull request:
Hi! New contributor to this project here.

This PR refactors error handling in the local/manager/server binaries, resulting in a net reduction of 26 lines of code.

ClaytonKnittel commented:
I'm unsure if this is a breaking change, this is modifying the signatures of `pub` functions in `src/` and removing the `pub const` error codes. Feedback is welcome!

zonyitoo commented:
LGTM. Please add doc comments for structs, methods and the mod itself.

zonyitoo commented:
And what's the actual use-case of the error type? Do you actually use it as a library?

ClaytonKnittel commented:
Done.

The error type is not intended to be used outside of the code I've modified. I've changed the package visibility of mod error to package-local.

zonyitoo commented:
@ClaytonKnittel Please fix windows build.

ClaytonKnittel commented:
Fixed.
